config/filesystems.php

```diff
@@ -46,6 +46,16 @@
         'local' => [
             'driver' => 'local',
             'root'   => storage_path('app'),
+            'permissions' => [
+                'file' => [
+                   'public' => 0644,
+                   'private' => 0644,
```

Comment:
Are you sure setting "private" permissions to be accessible by the public is the right way to go? I don't know where and how these are used with Pixelfed, but it doesn't look like a good pattern. (Same goes for the directory permissions.)

Reply:
Good morning, 
Yes I have checked it. But I think you can remove the "public" parameter, since you are only using/respecting the "private". 
I have left both, in case I missed something.
Regards

Reply:
Also, as in my case I am running PHP under PHP-FPM, the user should always create the files and directories with permissions 0644/0755 respectively, otherwise the user running the WEB service will not be able to access these files generating a 403 error.

Reply:
> Also, as in my case I am running PHP under PHP-FPM, the user should always create the files and directories with permissions 0644/0755 respectively, otherwise the user running the WEB service will not be able to access these files generating a 403 error.

I am also attempting to run Pixelfed like this, and this patch solves the same problem for me. However, I don't know enough about PHP or Pixelfed to opine on whether this opens any security vulnerabilities.

Reply:
No security holes opens with this fix. The permissions 0755 and 644 are the commons use in web content.

Reply:
Everything else - including the files - are created with these perms already, so it makes no sense for the parent directories not to be. Current behaviour breaks any attempt at separating webserver from php-fpm.